library/math/numbersLine/src/components/toolbar/IconsToolbar.tsx

```diff
@@ -50,53 +49,39 @@ const IconsToolbar = ({ typeAction, iconUrl, isDragged, isMenu }: IProps) => {
   // Effect to update the icon source depending on various state changes:
   // isOpen: Indicates whether the toolbar menu is open or closed.
   // isHovered: Indicates whether the mouse is currently hovering over the toolbar icon.
-  // openRestartDialog: Indicates whether the restart dialog is open.
-  // dragElements: Array of draggable elements.
-  // visitableDisplayButton: Indicates the visibility status of the display button.
-  // color: The current drawing color.
-  // drawSituation: The current drawing situation (e.g., whether the canvas is empty or not).
+  // openRestartDialog: Indicates whether the restart dialog is open - for Restrt.
```

Comment:
I think you should separate this component to 5 components/
it look so messy the if and check every time what type ou are and its complicated components. I think it is not smart to unite them, you can use common function by hook or something